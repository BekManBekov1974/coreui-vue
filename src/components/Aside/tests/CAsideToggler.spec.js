import { mount } from "@vue/test-utils";
import Component from "../CAsideToggler";

const ComponentName = 'CAsideToggler'
const wrapper = mount(Component)

describe(`${ComponentName} .vue`, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  // Inspect the raw component options
  it('has toggle method', () => {
    expect(typeof Component.methods.toggle).toBe('function')
  })
  it('has asideToggle method', () => {
    expect(typeof Component.methods.asideToggle).toBe('function')
  })
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.is('button')).toBe(true)
    expect(wrapper.classes()).toContain('navbar-toggler')
  })
});
