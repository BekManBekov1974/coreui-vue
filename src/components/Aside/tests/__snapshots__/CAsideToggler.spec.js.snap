// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CAsideToggler .vue renders correctly 1`] = `
<button
  class="navbar-toggler"
  display="lg"
  type="button"
>
  <span
    class="navbar-toggler-icon"
  />
</button>
`;
